tests/VolleyManagement.Specs/TeamsContext/ManageTeamPlayersSteps.cs

```diff
@@ -1,87 +1,181 @@
+            var firstName = playerName.Substring(0, whitespaceCharIndex);
+            var lastName = playerName.Substring(whitespaceCharIndex + 1);
+
+            var _newPlayer = new PlayerEntity {
+                FirstName = firstName,
+                LastName = lastName
+            };
+
+            TestDbAdapter.CreatePlayer(_newPlayer);
+            TestDbAdapter.AssignPlayerToTeam(_newPlayer.Id, _teamId.Id);
+            _roster.Add(new PlayerId(_newPlayer.Id));
         }
-        
-        [Given(@"John Smith is a team player")]
-        public void GivenJohnSmithIsATeamPlayer()
+
+        [Given(@"I have removed (.*)")]
+        [Scope(Feature = "Manage Team Players")]
+        public void GivenIHaveRemovedPlayer(string playerName)
         {
-            ScenarioContext.Current.Pending();
+            var whitespaceCharIndex = playerName.IndexOf(' ');
+            var firstName = playerName.Substring(0, whitespaceCharIndex);
+            var lastName = playerName.Substring(whitespaceCharIndex + 1);
+
+            using (var ctx = TestDbAdapter.Context)
+            {
+                var playerEntities = ctx.Players
+                    .SingleOrDefault(p => p.FirstName == firstName && p.LastName == lastName);
+                _playerToRemove = new PlayerId(playerEntities.Id);
+            }
         }
-        
-        [Given(@"Jane Doe is a team captain")]
-        public void GivenJaneDoeIsATeamCaptain()
+
+        [Given(@"(.*) is a team captain")]
+        [Scope(Feature = "Manage Team Players")]
+        public void GivenJaneDoeIsATeamCaptain(string playerName)
         {
-            ScenarioContext.Current.Pending();
+            var whitespaceCharIndex = playerName.IndexOf(' ');
+            var firstName = playerName.Substring(0, whitespaceCharIndex);
+            var lastName = playerName.Substring(whitespaceCharIndex + 1);
+
+            using (var ctx = TestDbAdapter.Context)
+            {
+                var playerEntity = ctx.Players
+                    .SingleOrDefault(p => p.Id == _captainId.Id);
+
+                if (playerEntity == null)
+                {
+                    return;
+                }
+
+                playerEntity.FirstName = firstName;
+                playerEntity.LastName = lastName;
+                ctx.SaveChanges();
+            }
         }
-        
+
         [When(@"I execute AddPlayersToTeam")]
         public void WhenIExecuteAddPlayersToTeam()
         {
-            ScenarioContext.Current.Pending();
+            _teamService.AddPlayers(_teamId, _playerToAdd);
         }
-        
+
         [When(@"I execute RemovePlayersFromTeam")]
         public void WhenIExecuteRemovePlayersFromTeam()
         {
-            ScenarioContext.Current.Pending();
+            try
+            {
+                _teamService.RemovePlayers(_teamId,
+                    _playerToRemove != null ? new List<PlayerId> { _playerToRemove } : _roster);
```

Comment:
Can it be null?

Reply:
Yes. For example in this scenario. Our  _playerToRemove variable is null, becouse there were no steps to remove somebody from roster.
![image](https://user-images.githubusercontent.com/28676005/39574079-7f9213dc-4ede-11e8-9417-595e616479ad.png)

Reply:
@sdiachen or there is mistake on 32 and 33 lines. I think we should "remove" instead of "add player"